lib/ansible/modules/network/dnac/dnac_rest.py

```diff
@@ -0,0 +1,239 @@
+        required: true
+    method:
+        description:
+            - The HTTP method of the request.
+            - Using C(delete) is typically used for deleting objects.
+            - Using C(get) is typically used for querying objects.
+            - Using C(post) is typically used for modifying objects.
+        required: true
+        choices: [ delete, get, post ]
+    path:
+        description:
+            - Directory path to the endpoint. Do not include FQDN specified in C(host).
+        required: true
+    timeout:
+        description:
+            - HTTP timeout value.
+        default: 30
+    use_proxy:
+        description:
+            - If C(no), it will not use a proxy, even if one is defined in an environment variable on the target hosts.
+        type: bool
+        default: 'false'
+    use_ssl:
+        description:
+            - If C(no), it will use HTTP. Otherwise it will use HTTPS.
+        type: bool
+        default: 'true'
+    validate_certs:
+        description:
+            - If C(no), HTTPS certificates will not be verified.
+        type: bool
+        default: 'true'
+    content:
+        description:
+            - Raw content which should be fed in body.
+
+author:
+    - Kevin Breit (@kbreit)
+'''
+
+EXAMPLES = '''
+# Query inventory
+- name: Query network device inventory
+  sda_rest:
+    username: dnacuser
+    password: dnaccpass
+    host: dnac.yourorg.com
+    method: get
+    path: '/api/v1/network-device'
+    use_ssl: Yes
+  delegate_to: localhost
+
+'''
+
+RETURN = '''
+message:
+    description: Data returned from controller.
+    type: dict
+    returned: info
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url, open_url
+from ansible.module_utils.basic import json
+from ansible.module_utils._text import to_native
+
+
+def run_module():
+
+    # define the available arguments/parameters that a user can pass to
+    # the module
+    module_args = dict(
+        username=dict(type='str', required=True),
+        password=dict(type='str', required=True, no_log=True),
+        host=dict(type='str', required=True),
+        method=dict(type='str', choices=['delete', 'get', 'post'], required=True),
+        path=dict(type='path', required=True),
+        timeout=dict(type='int', default=30, required=False),
+        use_proxy=dict(type='bool', default=False, required=False),
+        use_ssl=dict(type='bool', default=True, required=False),
+        validate_certs=dict(type='bool', default=True, required=False),
+        content=dict(type='raw', required=False)
+    )
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # change is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+        message=''
```

Comment:
In python we tend to add a trailing comma, also for the last item in a multi-line dict or list.
This is explicitly allowed by PEP8 and ensures that if any items are added, only one new line is added (and not the previous line needs a change). So it keeps the origin of lines (blame) clean.

And it's also much more convenient.